src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -94,12 +96,38 @@ private function executeGuardAuthenticator($uniqueGuardKey, GuardAuthenticatorIn
                 $this->logger->debug('Calling getCredentials() on guard configurator.', array('firewall_key' => $this->providerKey, 'authenticator' => get_class($guardAuthenticator)));
             }
 
+            // abort the execution of the authenticator if it doesn't supports the request.
+            if ($guardAuthenticator instanceof GuardAuthenticatorInterface) {
+                if (true !== $guardAuthenticator->supports($request)) {
+                    return;
+                }
+                // as there was a support for given request,
+                // authenticator is expected to give not-null credentials.
+                $credentialsCanBeNull = false;
+            } else {
+                // it's a GuardAuthenticatorLegacyInterface
+                // we support the previous behaviour to avoid BC break.
+                $credentialsCanBeNull = true;
+                @trigger_error('The Symfony\Component\Security\Guard\GuardAuthenticatorInterface interface is deprecated since version 3.0 and will be removed in 4.0. Use Symfony\Component\Security\Guard\Authenticator\GuardAuthenticatorInterface instead.', E_USER_DEPRECATED);
```

Comment:
Could be
```php
@trigger_error(sprintf'The "%s" interface is deprecated since version 3.1 and will be removed in 4.0. Use "%s" instead.', GuardAuthenticatorLegacyInterface::class, GuardAuthenticatorInterface::class), E_USER_DEPRECATED);